tasks/src/main/scala/dagr/tasks/fgbio/FgBioTask.scala

```diff
@@ -0,0 +1,63 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package dagr.tasks.fgbio
+
+import java.nio.file.Path
+
+import dagr.core.config.Configuration
+import dagr.core.tasksystem.{FixedResources, ProcessTask}
+import dagr.tasks.JarTask
+
+import scala.collection.mutable.ListBuffer
+
+
+object FgBioTask {
+  val FgBioJarConfigPath = "fgbio.jar"
+}
+
+/**
+  * Base Task for any task in the FgBio jar.
+  */
+abstract class FgBioTask extends ProcessTask with JarTask with FixedResources with Configuration {
+
+  /** Looks up the first super class that is does not have "\$anon\$" in its name. */
+  def commandName: String = JarTask.findCommandName(Option(getClass), Some("FgBioTask"))
```

Comment:
why make this a `def` that gets invoked every time you need the command name (e.g. in the `args` method), and not just either `[lazy] val commandName`, or even just `this.name = JarTask.....`?